--- origsrc/libgcrypt-1.4.5/src/Makefile.am	2009-04-02 04:25:32.000000000 -0500
+++ src/libgcrypt-1.4.5/src/Makefile.am	2010-07-12 19:49:22.625909800 -0500
@@ -52,7 +52,7 @@ libgcrypt_la_SOURCES = g10lib.h visibili
 
 if HAVE_W32_SYSTEM
 
-LTRCCOMPILE = $(LIBTOOL) --mode=compile $(RC) \
+LTRCCOMPILE = $(LIBTOOL) --mode=compile --tag=RC $(RC) \
      `echo $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) | \
      sed -e 's/-I/--include-dir /g;s/-D/--define /g'`
 
@@ -66,6 +66,7 @@ no_undefined = -no-undefined
 export_symbols = -export-symbols $(srcdir)/libgcrypt.def
 
 install-def-file:
+	$(MKDIR_P) $(DESTDIR)$(libdir)
 	$(INSTALL) $(srcdir)/libgcrypt.def $(DESTDIR)$(libdir)/libgcrypt.def
 
 uninstall-def-file:
--- origsrc/libgcrypt-1.4.5/src/gcrypt.h	2009-12-11 09:43:51.000000000 -0600
+++ src/libgcrypt-1.4.5/src/gcrypt.h	2010-07-12 19:40:58.861096100 -0500
@@ -37,7 +37,7 @@
 # include <sys/socket.h>
 #endif /*!_WIN32*/
 
-typedef socklen_t gcry_socklen_t;
+typedef int gcry_socklen_t;
 
 #include <sys/time.h>
 
